src/cluster/Telemetry.cc

```diff
@@ -131,28 +127,41 @@ void VerboseTelemetry::OnIncomingEvent(std::string_view topic, std::string_view
 
 namespace {
 
-std::string determine_script_location() {
-    std::string result = "none";
-
-    if ( zeek::detail::call_stack.empty() )
-        return result;
+// Cache CallExpr pointers to their location.
+std::map<const zeek::detail::CallExpr*, std::string> location_cache;
 
+// Cached lookup of a script location.
+std::string_view determine_script_location() {
     ssize_t sidx = static_cast<ssize_t>(zeek::detail::call_stack.size()) - 1;
-
     while ( sidx >= 0 ) {
         const auto* func = zeek::detail::call_stack[sidx].func;
         const auto* ce = zeek::detail::call_stack[sidx].call;
 
-        // without_zeekpath_component looks pretty expensive and might be
-        // better to cache the result using the ce pointer instead of computing
-        // it over and over again.
+        // Cached?
+        auto it = location_cache.find(ce);
+        if ( it != location_cache.end() )
+            return it->second;
+
+        // Future: Ignore wrapper functions if we ever come across some.
+        // We only care about Broker::publish() and Cluster::publish() and
+        // these aren't wrapped, so currently nothing to do here.
+        //
+        // if ( ignore func ) {
+        //     --sidx;
+        //      continue;
+        // }
```

Comment:
At least to me it is not clear when something needs to happen here, should we just drop this comment?

Reply:
I'm going to keep this. Let me extend the comment to talk about `Func.cc::emit_builtin_error_common()` which does some magic in that area.